src/filesystem/interface.js

```diff
@@ -206,6 +206,60 @@ function FileSystem(options, callback) {
     return watcher;
   };
 
+  //Object that uses filenames as keys
+  const statWatchers = new Map();
```

Comment:
Sorry to flip back and forth on this, but I'm not clear what benefit we're gaining from `Map` here over `{}` for potential loss in compatibility.  Can we switch back to `{}` instead?